### PR TITLE
Fix circleci yaml

### PR DESCRIPTION
**Pull request by ksunden**
## PR summary

Follow up to #28425 and #28413

YAML is sometimes confusing and puts the ending `\n` on a multiline string with `|`, which made the job fail.

I moved the comment, hopefully it doesn't mess with dependabot being able to automatically update it, but the line was still too long with the comment on the same line.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: .github/workflows/circleci.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ jobs:
     name: Run CircleCI artifacts redirector
     steps:
       - name: GitHub Action step
-        uses: | #v1.0.0
+        uses:
```
Comment (QuLogic) [inline .github/workflows/circleci.yml:13]: We want to remove all whitespace:
```suggestion
        uses: >-
```

Comment (ksunden) [reply .github/workflows/circleci.yml:13]: According to yaml converters, looks like what is there _should_ work, not sure we need to change it, though `>-` would have also been viable to strip the whitespace.

**Comment by QuLogic** [comment]
Just wondering how this works with dependabot if the version is elsewhere?

**Comment by rcomer** [comment]
We _could_ just increase the max line length to 119...

**Comment by ksunden** [comment]
For reference, Dependabot appears to only look at inlined comments following the SHA hash:

https://github.com/dependabot/dependabot-core/blob/b78e6194fb6f8d609a0ded3f0c367962fcd946c6/github_actions/lib/dependabot/github_actions/file_updater.rb#L81

Examples from their tests can be found in:

https://github.com/dependabot/dependabot-core/blob/b78e6194fb6f8d609a0ded3f0c367962fcd946c6/github_actions/spec/fixtures/workflow_files/pinned_sources_version_comments.yml#L7


(Mostly leaving this as a future reference for when this comes up again... who knows if I'll think to find this, but would rather have it if I can)

**Review by QuLogic — APPROVED** [review]

**Comment by rcomer** [comment]
Added the backport commit to the existing #28426

**Pull request merged**